pyart/util/columnsect.py

```diff
@@ -0,0 +1,294 @@
+    # convert latitude/longitudes to radians
+    rad_lat = rad_lat * (np.pi/180.)
+    tar_lat = tar_lat * (np.pi/180.)
+    rad_lon = rad_lon * (np.pi/180.)
+    tar_lon = tar_lon * (np.pi/180.)
+    # difference in latitude
+    d_lat = (tar_lat - rad_lat)
+    # difference in longitude
+    d_lon = (tar_lon - rad_lon)
+    # Haversine formula
+    numerator = ((np.sin(d_lat/2.0)**2.0)
+                 + np.cos(rad_lat) * np.cos(tar_lat)
+                 * (np.sin(d_lon/2.0)**2.0))
+    distance = 2 * 6371000 * np.arcsin(np.sqrt(numerator))
+
+    # return the output
+    return distance
+
+
+def for_azimuth(rad_lat, tar_lat, rad_lon, tar_lon):
+
+    """
+    Calculation of inital bearing along a great-circle arc
+    Known as Forward Azimuth Angle.
+
+    Assumptions
+    -----------
+    Radius of the Earth = 6371 km / 6371000 meters
+    Distance is calculated for a smooth sphere
+    Radar and Target are at the same altitude (need to check)
+
+    Parameters
+    ----------
+    rad_lat : float, [degrees]
+        latitude of the radar in degrees
+    tar_lat : float, [degrees]
+        latidude of the target in degrees
+    rad_lon : float, [degrees]
+        longitude of the radar in degrees
+    tar_lon : float, [degrees]
+        longitude of the target in degress
+
+    Returns
+    -------
+    azimuth : float, [degrees]
+        azimuth angle from the radar where
+        target is located within the scan.
+        output is in degrees.
+    """
+    # check the input lat/lon
+    check_lat(rad_lat)
+    check_lat(tar_lat)
+    check_lon(rad_lon)
+    check_lon(tar_lon)
+    # convert latitude/longitudes to radians
+    rad_lat = rad_lat * (np.pi/180.)
+    tar_lat = tar_lat * (np.pi/180.)
+    rad_lon = rad_lon * (np.pi/180.)
+    tar_lon = tar_lon * (np.pi/180.)
+    # Differnce in longitudes
+    d_lon = tar_lon - rad_lon
+    # Determine x,y coordinates for arc tangent function
+    corr_y = np.sin(d_lon) * np.cos(tar_lat)
+    corr_x = ((np.cos(rad_lat) * np.sin(tar_lat))
+              - (np.sin(rad_lat) * np.cos(tar_lat) * np.cos(d_lon)))
+    # Determine forward azimuth angle
+    azimuth = np.arctan2(corr_y, corr_x) * (180./np.pi)
+
+    # Return the output as a function of 0-360 degrees
+    if azimuth < 0:
+        azimuth += 360.
+
+    return azimuth
+
+
+def get_fields_latlon(radar, lat, lon):
+
+    """
+    Given the location (in lat,lon) of a target, extract the radar column
+    above that point for further analysis.
+
+    Parameters
+    ----------
+    radar : Radar Object
+        Py-ART Radar Object from which distance to the target, along
+        with gates above the target, will be calculated.
+    lat : float, [degrees]
+        Latitude, in degrees North, of the target.
+    lon : float, [degrees]
+        Longitude, in degrees East, of the target.
+
+    Function Calls
+    --------------
+    sphere_distance
+    for_azimuth
+    get_column_rays
+
+    Returns
+    -------
+    column : dict
```

Comment:
Can you have it return a radar object? That would make it easier to plot/such... or an Xarray.Dataset?